src/components/timeline/Tooltip.svelte

```diff
@@ -76,35 +77,98 @@
     }
   }
 
-  function tooltipText() {
-    let text = '';
-    if (points.length) {
-      text += tooltipTextForPoints();
+  function tooltipText(): string {
+    let tooltipText = '';
+
+    activityDirectives.forEach((activityDirective: ActivityDirective, i: number) => {
+      const text = textForActivityDirective(activityDirective);
+      tooltipText = `${tooltipText} ${text}`;
+
+      if (i !== activityDirectives.length - 1) {
+        tooltipText = `${tooltipText}<hr>`;
+      }
+    });
+
+    if (constraintViolations.length && activityDirectives.length) {
+      tooltipText = `${tooltipText}<hr>`;
     }
-    if (violations.length) {
-      if (points.length) {
-        text += '<hr>';
+
+    constraintViolations.forEach((constraintViolation: ConstraintViolation, i: number) => {
```

Comment:
When hovering over a span where there are also constraint directives the divider appears to be missing: 
![image](https://user-images.githubusercontent.com/4419607/223511476-95732eca-1213-4ce3-8b7f-8bf3c62b0f41.png)

Reply:
Fixed!